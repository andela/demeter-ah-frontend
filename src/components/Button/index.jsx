import React from 'react';

const Button = ({
  onClick, classes, type, name, children, isSubmit, ...rest
}) => (
  <button
    onClick={onClick}
    className={`btn ${classes}`}
    type={type}
    disabled={isSubmit || false}
    {...rest}
  >
    { !isSubmit ? (children || name) : 'Loading...'}
  </button>
);

export default Button;
