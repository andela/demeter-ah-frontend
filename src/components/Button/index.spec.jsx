
import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './index';
// import InputForm from '../InputForm';

describe('Button Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });

  it('should display Loading... onSubmit prop function when form is submitted', () => {
    const state = true;
    const component = mount(<Button name="test" isSubmit={state} />);
    const res = component.find('button').at(0).text();
    expect(res).toEqual('Loading...');
  });
});
