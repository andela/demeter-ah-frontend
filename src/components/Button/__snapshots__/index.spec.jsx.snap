// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render without errors 1`] = `
<button
  className="btn undefined"
  disabled={false}
/>
`;
